README.md

```diff
@@ -13,3 +13,8 @@ recursive call considers.
 Analyse the time complexity of your implementation and give a $\Theta$ bound for
 its worst-case runtime. Add your answer, including your reasoning, to this
 markdown file.
+
+The best and average case for this algorthim is O(nlogn) because it takes logn to split the array and then it takes a constant time to partition giving nlogn. The worst case is n^2 though becuase if its already sorted than it is not split in half and will have to look at each element and then partition each giving n in both leading to worst case being O(n^2). By using the big O notation we can deduce that big theta will be the same with best and average being nlog n and the worst case being n^2.
+
+I had chat gpt help with how to write the partition fuction and the key ideas of making it iterative instead of recursive
```

Comment:
Can you be more specific here please? Did ChatGPT write the function for you?

Reply:
more specific about the run time? And I couldnt figure out how to start so i had chat gpt help with most of the fucntion yes.

Reply:
And what about the rest of the code? Did ChatGPT write that?

Reply:
If its to similar i can re write it now that i have a general idea of how it might work?

Reply:
I can't tell whether it's too similar if you don't tell me what ChatGPT generated...

Reply:
I cant remember fully but i believe i only wrote one or two lines of the code

Reply:
Then I'm afraid I can't count this.

Reply:
can i rewrite it, since the explanation was my own?

Reply:
Sure.

Reply:
Thank you for letting me retry this i have rewritten this except for the part i specified in the readme file.